src/controllers/portfolio/portfolio.ts

```diff
@@ -618,33 +637,33 @@ export class PortfolioController extends EventEmitter {
                 portfolioLib,
                 {
                   blockTag: 'pending',
-                  previousHints: storagePreviousHints[key],
+                  previousHints: fallbackHints,
                   ...(currentAccountOps && {
                     simulation: {
                       account: selectedAccount,
                       accountOps: currentAccountOps
                     }
                   }),
                   isEOA: !isSmartAccount(selectedAccount),
-                  additionalHints: this.#additionalHints
+                  additionalHints
                 },
                 forceUpdate
               )
             : Promise.resolve(false)
         ])
 
-        // Persist previousHints in the disk storage for further requests, when:
-        // latest state was updated successful and hints were fetched successful too (no HintsError from portfolio result)
-        if (
-          isSuccessfulLatestUpdate &&
-          !(accountState[network.id]!.result?.errors || []).find((err) => err.name === 'HintsError')
-        ) {
-          storagePreviousHints[key] = getHintsWithBalance(
-            accountState[network.id]!.result!
-            // !hasNonZeroTokens,
-            // this.#additionalHints
+        // Persist latest state in previousHints in the disk storage for further requests
+        if (isSuccessfulLatestUpdate && !areAccountOpsChanged && !forceUpdate) {
```

Comment:
A small thing, but `forceUpdate` equals `opts?.forceUpdate || areAccountOpsChanged`, defined a bit above. So removing `!areAccountOpsChanged` seems safe as it's included in `forceUpdate`. No problem to keep it just in case, though.

And please tell me later why we don't call `getUpdatedHints` when forceUpdate is true

Reply:
We need to update only on latest state, so I think its safe to check only `!areAccountOpsChanged` and this is a mistake on my end.